### ISSUE TITLE
volumesnaphot: jobs are not cleaned

### ISSUE THREAD

**Issue by sebastien-prudhomme** [stashed/stash#1089]
Version 0.9.0-rc6

When using  the VolumeSnapshotter driver, stash doesn't clean ```stash-vs-deploy-*``` jobs:

```
kubectl get job -n elasticsearch
NAME                                           COMPLETIONS   DURATION   AGE
stash-vs-sts-elasticsearch-master-1587942004   1/1           22s        15h
stash-vs-sts-elasticsearch-master-1587987055   1/1           31s        3h22m
stash-vs-sts-elasticsearch-master-1587987371   1/1           15s        3h17m
stash-vs-sts-elasticsearch-master-1587987723   1/1           24s        3h11m
stash-vs-sts-elasticsearch-master-1587989642   1/1           24s        159m
stash-vs-sts-elasticsearch-master-1587992406   1/1           81s        113m
stash-vs-sts-elasticsearch-master-1587996035   1/1           52s        53m
```

backupsessions are correctly cleaned (I use the default history of 1):

```
kubectl -n elasticsearch get backupsession
NAME                       INVOKER-TYPE          INVOKER-NAME    PHASE       AGE
elasticsearch-1587996035   BackupConfiguration   elasticsearch   Succeeded   54m
``` 

In the logs I can also notice that another job with a different name is created and deleted:

```
I0427 14:00:00.653618       1 jobs.go:68] Sync/Add/Update for Job stash-backup-elasticsearch-1587996000
I0427 14:00:00.785927       1 jobs.go:68] Sync/Add/Update for Job stash-backup-elasticsearch-1587996000
I0427 14:00:35.193301       1 backup_session.go:104] Sync/Add/Update for BackupSession elasticsearch-1587996035
I0427 14:00:35.227100       1 job.go:36] Creating Job elasticsearch/stash-vs-sts-elasticsearch-master-1587996035.
I0427 14:00:35.292000       1 backup_session.go:104] Sync/Add/Update for BackupSession elasticsearch-1587996035
I0427 14:00:35.298567       1 backup_session.go:147] Skipping processing BackupSession elasticsearch/elasticsearch-1587996035. Reason: phase is "Running".
I0427 14:00:35.745991       1 jobs.go:68] Sync/Add/Update for Job stash-backup-elasticsearch-1587996000
I0427 14:00:35.746112       1 jobs.go:71] Deleting succeeded job stash-backup-elasticsearch-1587996000
I0427 14:00:35.762164       1 jobs.go:82] Deleted stash job: stash-backup-elasticsearch-1587996000
W0427 14:00:35.762326       1 jobs.go:64] Job elasticsearch/stash-backup-elasticsearch-1587996000 does not exist anymore
```

**Comment by sebastien-prudhomme** [comment]
Job owner reference is this:

```
  ownerReferences:
  - apiVersion: stash.appscode.com/v1beta1
    blockOwnerDeletion: true
    controller: true
    kind: BackupConfiguration
    name: elasticsearch
```

but this comment says it should be backupsession and not backupconfiguration: https://github.com/stashed/stash/blob/master/pkg/controller/backup_session.go#L364

The code in ```ensureBackupJob``` function seems to be correct, not the one in ```ensureVolumeSnapshotterJob```

**Comment by sebastien-prudhomme** [comment]
For the second backup job, seems that there is a missing condition "invoker.Driver != api_v1beta1.VolumeSnapshotter" on this line: https://github.com/stashed/stash/blob/53dfe8b0ed5ae5482d69e83e57bede95ed26045d/pkg/controller/backup_session.go#L177

**Comment by hossainemruz** [comment]
@sebastien-prudhomme you are right. I believe this bug was introduced during refactoring the code to support batch backup. Can you please send a PR?

Also, this comment seems to be misplaced. It should be in line 299. Can you please fix that too. Also, add an appropriate comment for line 297.

**Comment by sebastien-prudhomme** [comment]
@hossainemruz i'm not a Go developer at all, i try to understand it but i've no knowledge of this language...

**Comment by hossainemruz** [comment]
Ok. No problem. We will fix it ASAP. Thank you for identifying the bug.

**Comment by hossainemruz** [comment]
This issue should be fix by #1090. Can you please confirm?

Use the latest image built from the fix.
```
helm install stash-operator appscode/stash \
  --version v0.9.0-rc.6 \
  --namespace kube-system \
  --set operator.registry=appscodeci \
  --set operator.tag=fix-vs-job-cleanup_linux_amd64
```

**Comment by sebastien-prudhomme** [comment]
@hossainemruz old jobs and their associated pods are deleted now. Thanks
